numba/tests/test_errorhandling.py

```diff
@@ -265,6 +266,23 @@ def call_intrin():
         self.assertIn(given_reason1, excstr)
         self.assertIn(given_reason2, excstr)
 
+    def test_error_function_source_is_correct(self):
+        """ Checks that the reported source location for an overload is the
+        overload implementation source, not the actual function source from the
+        target library."""
+
+        @njit
+        def foo():
+            np.linalg.svd("chars")
+
+        with self.assertRaises(errors.TypingError) as raises:
+            foo()
+
+        excstr = str(raises.exception)
+        self.assertIn(error_reasons['specific_error'].splitlines()[0], excstr)
+        expected_file = os.path.join("numba", "np", "linalg.py")
+        self.assertIn(expected_file, excstr)
```

Comment:
The test passes without the patch because the filename is in the traceback. Perhaps, check that `Overload in function 'svd_impl': ` in part of the line